README.md

```diff
@@ -6,32 +6,223 @@
 
 Convert between [itk](https://itk.org) and [napari](https://napari.org) data structures.
 
-Installation
-------------
+## Installation
 
-```
+```sh
 pip install itk-napari-conversion
 ```
 
-Usage
------
+## Usage
 
-Convert an `itk.Image` to an `napari.layers.Image`:
+### Image Conversion
 
-```
+#### Convert ITK Image to napari Image Layer
+
+Convert an `itk.Image` to a `napari.layers.Image`:
+
+```python
 from itk_napari_conversion import image_layer_from_image
 
 image_layer = image_layer_from_image(image)
 ```
 
-Convert to an `napari.layers.Image` to an `itk.Image`:
+**Features:**
+- Automatically detects and handles RGB/RGBA images
+- Preserves image metadata (spacing, origin, direction, custom metadata)
+- Converts ITK's physical space information to napari's layer transformations:
+  - `spacing` → `scale`
+  - `origin` → `translate`
+  - `direction` → `rotate`
+
+**Example:**
+```python
+import itk
+import napari
+from itk_napari_conversion import image_layer_from_image
+
+# Read an image
+image = itk.imread('path/to/image.nrrd')
+
+# Add custom metadata
+image['units'] = 'mm'
+image['patient_id'] = '12345'
+
+# Convert to napari layer
+image_layer = image_layer_from_image(image)
+
+# The layer will have:
+# - image_layer.scale = image spacing (in reverse order for NumPy)
+# - image_layer.translate = image origin (in reverse order for NumPy)
+# - image_layer.rotate = image direction matrix (in reverse order for NumPy)
+# - image_layer.metadata = all custom metadata
 ```
+
+#### Convert napari Image Layer to ITK Image
+
+Convert a `napari.layers.Image` to an `itk.Image`:
+
+```python
 from itk_napari_conversion import image_from_image_layer
 
 image = image_from_image_layer(image_layer)
 ```
 
-Hacking
+**Features:**
+- Automatically handles RGB/RGBA layers
+- Converts napari layer transformations back to ITK physical space:
+  - `scale` → `spacing`
+  - `translate` → `origin`
+  - `rotate` → `direction`
+- Preserves all metadata from the layer
+
+**Example:**
+```python
+import numpy as np
+import napari
+from itk_napari_conversion import image_from_image_layer
+
+# Create a napari image layer with transformations
+viewer = napari.Viewer()
+data = np.random.rand(100, 100, 100)
+layer = viewer.add_image(
+    data,
+    scale=[2.0, 1.5, 1.5],  # anisotropic spacing
+    translate=[10.0, 20.0, 30.0],
+    metadata={'description': 'My volume'}
+)
+
+# Convert to ITK
+image = image_from_image_layer(layer)
+
+# The ITK image will have:
+# - spacing: coordinates in ITK order, so reversed from napari `scale`: [1.5, 1.5, 2.0]
+# - origin: coordinates in ITK order, so reversed from napari `translate`: [30.0, 20.0, 10.0]
+# - direction: from napari `rotate`. In this case unspecified, so just the identity matrix.
```

Comment:
Can you possibly extend the example here with a non-identity rotation matrix:

```py
viewer.add_image(
    data,
    scale=[2.0, 1.5, 1.5],  # anisotropic spacing
    translate=[10.0, 20.0, 30.0],
    rotate=[ ... ],
```

And then possibly mention in the comment (line 100 here) what the ITK `direction` then would be?